docs/v1/P4Runtime-Spec.mdk

```diff
@@ -6304,6 +6324,380 @@ field [@ProtoAny]. At the moment, there is not any mechanism to extend the
+  type space is left up to the target. Some optional guidelines
+  * For P4 externs, any new extern to have a new table type.
+  * For non-P4 externs, options are
+    * All non-P4 tables grouped under one giant non_p4 table type
+    * Tables divided on the basis of categories. Categories are defined in the
+      spec but aren't part of the p4info.
+    * Every table has its own table-type. This can however lead to running out
+      of the 8 bit type ID space soon.
+  Note that this value does not need
+  to be unique across all architectures from all organizations, since at any
+  given time every device managed by a P4Runtime server maps to a single P4Info
+  message and a single architecture.
+
+* `generic_table_type_name`, which specifies the fully-qualified P4 name of the
+  generic_table in case of P4 entities. In cases of non-P4 entities, it is a
+  name which uniquely identifies this entity in the entire space including P4
+  objects
+
+* `repeated string generic_table_properties`, please check generic-table
+  properties under section [GenericTable categories]
+  (#sec-generic-table-categories) for more details
+
+* `repeated GenericTableInstance`, contains the info for one instance of
+  a generic-table-type. All tables of one type are grouped under this.
+
+GenericTableInstance
+* `preamble`, a `Preamble` message with the ID, name, and alias of this
+  GenericTable.
+
+* `generic_match_fields`, a repeated field of type `GenericMatchField`
+  representing the data to be used to construct the lookup key matched in this
+  table. For information check the generic_match_fields info in the [Generic
+  Table section](#sec-generic-table-entry)
+
+* `union_refs`, a repeated `UnionRef` field representing the set of possible
+  unions for this table. Functionally, it behaves same as that of ActionRefs.
+  Hence it has been kept as ActionRefs itself.
+  Please check the union_refs info in the [Generic Table section]
+  (#sec-generic-table-entry)
+
+* `const_default_union_id`, if this table has a constant default union, this
+  field will carry the `uint32` identifier of that action, otherwise its value
+  will be 0. A default union is the data when the key is null. It is similar to
+  a default action for Match Tables. Being constant means that the control
+  plane cannot set a different default union at runtime or change the default
+  union's arguments.
+
+* `size`, an `int64` describing the desired number of table entries that the
+  target should support for the table.  See the "Size" subsection within the
+  "Table Properties" section of the P4~16~ language specification for details
+  [@P4TableProperties].
+
+* `other_properties`, an `Any` Protobuf message [@ProtoAny] to embed
+  architecture-specific or target-specific table properties [@P4TableProperties]
+  that the target wants to convey.
+
+## `GenericData` { #sec-p4data-generic-data}
+
+`GenericData` is an addition to p4data.proto- which aims at adding support for
```

Comment:
"which aims at adding support" -> "which adds support".  If it isn't hitting the target, we should update it so it does :-)